pkg/cmd/infra/deployer/deployer.go

```diff
@@ -78,7 +78,7 @@ func NewCommandDeployer(name string) *cobra.Command {
 }
 
 // NewDeployer makes a new Deployer from a kube client.
-func NewDeployer(client kclient.Interface) *Deployer {
+func NewDeployer(client *kclient.Client) *Deployer {
```

Comment:
if it's to plumb what's needed for logging, can we take an interface with all the required info instead (namespace, name, and PodLogOptions)?